src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_TIMES_VALUE = 1.5;
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (data.length == this.size) {
+            resize();
+        }
+        data[this.size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
+        checkIndex(index);
+        if (data.length == this.size) {
+            resize();
+        }
+        for (int i = this.size - 2; i >= index; i--) {
+            data[i + 1] = data[i];
+        }
+        this.data[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) data[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        data[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        this.size--;
```

Comment:
the same for size here, you should decrement the size after removing the element